core-blocks/heading/edit.js

```diff
@@ -66,6 +67,13 @@ export default function HeadingEdit( {
 				onSplit={
 					insertBlocksAfter ?
 						( before, after, ...blocks ) => {
+							if ( ! canInsertSibling( 'core/paragraph' ) ) {
```

Comment:
Would there be any way to move this to `block-list` inside `insertBlocksAfter` to avoid having the block implementor care about this? I feel like this logic should be part of core, not the block.